File: DAV/tips_histogram.R
# Install and load necessary packages
if (!require(ggplot2)) install.packages("ggplot2")
if (!require(readr)) install.packages("readr")
if (!require(dplyr)) install.packages("dplyr")

library(ggplot2)
library(readr)
library(dplyr)

# Load tips dataset from GitHub
tips <- read_csv("https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv")

# Show top 10 records
cat("Top 10 Records:\n")
print(head(tips, 10))

# Plot faceted histogram of tip by day
ggplot(tips, aes(x = tip)) +
  geom_histogram(binwidth = 0.5, fill = "skyblue", color = "black") +
  facet_wrap(~day) +
  theme_minimal() +
  ggtitle("Histogram of Tip Amount by Day") +
  xlab("Tip Amount") +
  ylab("Count")

File: DAV/tips_histogram.py
import seaborn as sns
import matplotlib.pyplot as plt

# Load tips dataset
tips = sns.load_dataset("tips")

# Display top 10 records
print("Top 10 Records:")
print(tips.head(10))

# Plot histogram of tip for each day
g = sns.FacetGrid(tips, col="day", col_wrap=2, height=4)
g.map(sns.histplot, "tip", bins=10, color="skyblue", edgecolor="black")
g.set_titles("Day: {col_name}")
g.fig.suptitle("Histogram of Tip Amount by Day", y=1.05)
plt.show()
